torch/nn/quantized/modules/conv.py

```diff
@@ -205,13 +205,11 @@ def get_qconv(cls, mod, activation_post_process, weight_post_process=None):
         qconv.set_weight_bias(qweight, mod.bias)
         try:
             act_scale, act_zp = activation_post_process.calculate_qparams()
+            qconv.scale = float(act_scale)
```

Comment:
Sorry, why can't we remove the try/except here? can we do something like
```
if activation_post_process is not None:
    ...
else:
    ...
```

Reply:
because it does have a activation_post_process, it just raises an error when you try to calculate_qparams

https://github.com/pytorch/pytorch/blob/f455030931c30bfeaab7f79b3f4305000674d6f2/torch/ao/quantization/observer.py#L1215

Reply:
I think if we want a try-except here, the placeholder observer could throw a more specific exception type and this except clause would catch just that type.

I'd vote +1 on removing the try-catch cause, it feels cleaner to never insert an incorrect observer in the first place.  We should make logic to check that observers are correct in the prepare pass, not in the convert pass.

Reply:
There is some logic here (https://github.com/pytorch/pytorch/blob/master/torch/ao/quantization/qconfig.py#L247) for checking whether a qconfig is valid for a module, perhaps this could be extended to cover this case.

Reply:
I edited it to check for activation post process dtype, which feels like it should be sufficient.